armi/reactor/converters/axialExpansionChanger.py

```diff
@@ -850,13 +850,23 @@ def determineTargetComponent(self, b, flagOfInterest=None):
         else:
             componentWFlag = [c for c in b.getChildren() if c.hasFlags(flagOfInterest)]
         if len(componentWFlag) == 0:
-            raise RuntimeError("No target component found!\n   Block {0}".format(b))
+            # didn't match flags so check type before failing
+            componentWFlag = [
+                c for c in b if (c.p.type in b.p.type or b.p.type in c.p.type)
+            ]
```

Comment:
I'm not sure I understand what the underlying assumption is here that makes this a valid thing to do. Basically you're saying that if a component in the block has the same name as the block itself then it should be the target component?

Reply:
It might be good to consider components by name since there is no guarantee that this will work for all block types and might be fragile. Is it worth considering a refactor for always using a component name or first flags and then name?

Reply:
After some offline discussions, this type check was rolled back in https://github.com/terrapower/armi/pull/1230/commits/d9d138b9b40264ba9e82639e810235d67f12175d.